src/full/Agda/Syntax/Parser.hs

```diff
@@ -149,7 +149,7 @@ parseLiterateWithComments p layers = do
   forM_ (map fst overlaps) $ \c ->
     warning $ OverlappingTokensWarning { warnRange = getRange c }
 
-  (, coh) <$> (return $ forMaybe terms $ \case
+  (, coh) <$> return (forMaybe terms $ \case
```

Comment:
This can even improved to 
```haskell
return $ (, coh) $ forMaybe terms $ \case ...
-- delete final )
```
to get rid of the long-ranging parenthesis as well.

Reply:
I did this. If I get rid of all ignores in `.hlint.yaml` and run hlint on this file:

```
hlint src/full/Agda/Syntax/Parser.hs
src/full/Agda/Syntax/Parser.hs:(152,3)-(156,48): Warning: Redundant <$>
Found:
  (, coh)
    <$>
      return
        (forMaybe terms
           $ \case
               Left t -> Just t
               Right (Layer Comment interval s) -> Just $ TokTeX (interval, s)
               Right (Layer Markup interval s) -> Just $ TokMarkup (interval, s)
               Right (Layer Code _ _) -> Nothing)
Perhaps:
  return
    ((, coh)
       (forMaybe terms
          $ \case
              Left t -> Just t
              Right (Layer Comment interval s) -> Just $ TokTeX (interval, s)
              Right (Layer Markup interval s) -> Just $ TokMarkup (interval, s)
              Right (Layer Code _ _) -> Nothing))

1 hint
```

Hlint only goes [one step at a time](https://github.com/ndmitchell/hlint#why-are-hints-not-applied-recursively) with its suggestions. Running it again, I get another suggestion that seems reasonable so I followed that one too:

```
$ hlint src/full/Agda/Syntax/Parser.hs
src/full/Agda/Syntax/Parser.hs:(152,12)-(156,48): Suggestion: Evaluate
Found:
  (, coh)
    (forMaybe terms
       $ \case
           Left t -> Just t
           Right (Layer Comment interval s) -> Just $ TokTeX (interval, s)
           Right (Layer Markup interval s) -> Just $ TokMarkup (interval, s)
           Right (Layer Code _ _) -> Nothing)
Perhaps:
  (forMaybe terms
     $ \case
         Left t -> Just t
         Right (Layer Comment interval s) -> Just $ TokTeX (interval, s)
         Right (Layer Markup interval s) -> Just $ TokMarkup (interval, s)
         Right (Layer Code _ _) -> Nothing, 
   coh)

1 hint
```

Reply:
Mmh, `hlint` isn't on the same priorities as I am.  I wish to eliminate long-ranging parentheses.  Basically, I want to avoid a parentheses that closes only several lines after it opened.  Larger structuring should be done by indentation, not parentheses, this is my opinion.

This is why I suggested
> ```haskell
> return $ (, coh) $ forMaybe terms $ \case ...
> -- delete final )
> ```
> 
> to get rid of the long-ranging parenthesis